### PR TITLE
 #161291012 Build Like/Dislike article Functionality

### PR DESCRIPTION
**Pull request by Darthrighteous**
#### What does this PR do?
Build the end point to like/dislike an article

#### Description of Task to be completed?
- Have the endpoint `POST /articles/:slug/:action` to like or dislike an article
- Ensure that the user can only like or unlike an article only once

#### How should this be manually tested?
- clone this repo, cd into it and checkout the branch `feature/161291012/user-like-dislike-article`
- set up your .env variables and local database
- run npm install to install required packages
- run npm start to start up the server
- use postman(or any api testing tool) to test the following routes
- first sign up a user using POST `http://localhost:{PORT}/api/v1/users/signup` getting the valid credentials
- then test the route using `POST http://localhost:{PORT}/api/v1/articles/:slug/:action`
  where `:slug` is the article slug and `:action` can be `like` or `dislike`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
 [161291012](https://www.pivotaltracker.com/story/show/161291012)

#### Screenshots (if appropriate)
screen shot 2018-11-12 at 10 43 40 am
screen shot 2018-11-12 at 11 11 56 am
screen shot 2018-11-12 at 11 12 05 am

#### Questions:
~**_Problem:_** I am currently using the article slug on this route to get the article. This means in the controller, I have to first query the article table to get the article's id then work with that. This doesn't seem optimal to me. I assume we are going to be querying with the `slug` on the front end and not the `id`.~

~**_Proposed Solution:_** so my proposed solution will be to create a middleware function that resolves the slug into an id, what do you guys think of this?~

### CODE REVIEW

File: server/controllers/ArticleController.js [hunk 1]
```diff
@@ -119,6 +119,97 @@ class ArticleController {
...
+      if (undoRows > 0) {
+        return res.status(200).json({
+          status: 'success',
+          message: `article ${suffix}, undo successful`,
+        });
+      }
+    };
+
+    // error for unknown action in switch statement
+    // eslint-disable-next-line max-len
```
Comment (sulenchy) [inline server/controllers/ArticleController.js]: I observe you disabled the max-len rule. The max len rule in the eslintrc file is to ensure our codebase is readable. Can you justify why you disable it here?

File: server/middlewares/verifyArticleId.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import models from '../models';
+
+const { Article } = models;
+
+const verifyArticleId = async (req, res, next) => {
```
Comment (sulenchy) [inline server/middlewares/verifyArticleId.js]: I feel `verifyArticleId` is not descriptive enough. When I see it for the first time, I thought you would query for the articleId but that's not the case. `validateArticleId` sounds better to me. What do you think?

File: server/controllers/ArticleController.js (hunk 1 above)
Comment (Darthrighteous) [reply server/controllers/ArticleController.js]: @sulenchy I agree that the reason for linting is to make our code more readable. However, I have disabled the `max-len` rule in this one line for the following reason:
- Following the airbnb style guide, it states not to write long strings over multiple lines to make them easier to work with. see here —> https://github.com/airbnb/javascript#strings.

File: server/middlewares/verifyArticleId.js (hunk 2 above)
Comment (Darthrighteous) [reply server/middlewares/verifyArticleId.js]: @sulenchy Thanks for pointing that out. I agree, validate is more descriptive of what the function does.

**Pull request merged**